### PR TITLE
CORE-537: Save ERC20 Tokens Persistently

### PR DESCRIPTION
**Pull request by EBGToo**
This had been on the back-burner.. but then CORE-554 happened.

I observed that in ewmCreate() there is this:
```
            FOR_SET (BREthereumLog, log, logs)
                ewmSignalLog (ewm, BCS_CALLBACK_LOG_ADDED, log);
```
signaling all logs.  And then in the handler there is this:
```
    BREthereumToken token = ewmLookupToken (ewm, logGetAddress(log));
    if (NULL == token) { logRelease(log); return;}
```
throwing away all the logs.

And then we wait for the next API sync to restore the logs after having installed the tokens.

This change moves the global `tokenSet` into `EWM` - a big win alone.  Tokens will exist immediately, from persistent storage, on `ewmCreate()`.  And then logs will be added to their appropriate wallet.  The balance is not updated based on the logs - that is in CORE-554.

### CODE REVIEW

File: ethereum/contract/BREthereumToken.c [hunk 1]
```diff
@@ -117,6 +117,59 @@ tokenGetContract(BREthereumToken token) {
...
+extern void
+tokenUpdate (BREthereumToken token,
+             const char *symbol,
+             const char *name,
+             const char *description,
+             int decimals,
+             BREthereumGas defaultGasLimit,
+             BREthereumGasPrice defaultGasPrice) {
+
+    if (0 != strcasecmp (symbol     , token->symbol     )) { free (token->symbol     ); token->symbol      = strdup (symbol     ); }
```
Comment (michael-brd) [inline ethereum/contract/BREthereumToken.c]: General question; can these (name, description, symbol) be unicode? The standard `strxyz` don't support multibyte encodings, IIRC.

Comment (EBGToo) [reply ethereum/contract/BREthereumToken.c]: I've not thought about it.  Worthy of a JIRA to capture?

File: ethereum/ewm/BREthereumEWM.c [hunk 2]
```diff
@@ -2955,3 +3051,48 @@ ewmTransferDelete (BREthereumEWM ewm,
...
+
+    if (NULL == token) {
+        token = tokenCreate (address,
+                             symbol,
+                             name,
+                             description,
+                             decimals,
+                             defaultGasLimit,
+                             defaultGasPrice);
+        BRSetAdd (ewm->tokens, token);
```
[the remark this answers is not in the document]
Comment (EBGToo) [reply ethereum/ewm/BREthereumEWM.c]: "Inconsistent Threading Model"

[the remark this answers is not in the document]
Comment (EBGToo) [reply ethereum/ewm/BREthereumEWM.c]: I added the `ewm->lock` over `ewm->tokens` *and* the update of the token itself.

File: ethereum/contract/BREthereumToken.c (hunk 1 above)
Comment (michael-brd) [reply ethereum/contract/BREthereumToken.c]: Absolutely.

**Pull request merged**